x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/mapper/SemanticTextFieldMapper.java

```
@@ -1296,6 +1307,11 @@ private static void configureDenseVectorMapperBuilder(
         MinimalServiceSettings modelSettings,
         SemanticTextIndexOptions indexOptions
     ) {
+        if (indexVersionCreated.before(INDEXED_BY_DEFAULT_INDEX_VERSION)) {
+            deprecationLogger.warn(DeprecationCategory.MAPPINGS, "semantic_text", WARNING_MESSAGE_8X);
+        }
+        denseVectorMapperBuilder.indexed(true);
```

💭❓  Setting indexed true in versions before `indexed_by_default`, might leave similiarity null. If it's not provided [below](https://github.com/elastic/elasticsearch/pull/133080/files#diff-bd2dcbac19a163cd148c39c6bb7b428bc1e67f85276e8c89ae2fb2b66399bce9R1316) (through the model settings) the builder will fail with 
```
  java.lang.IllegalArgumentException: Field [index] requires field [similarity] to be configured and not null
```

Do we want to add a check inside the if clause above and fail early(ier?) or is this acceptable?

---

Guess this won't be needed after all. After further discussion we are going to only support semantic_text creation for 8.11+ so both indexed and similarity will have populated defaults 👍 . Thanks for the discussion everyone